### ISSUE TITLE
Support for arranging shapes (e.g. layers)

### ISSUE THREAD

**Issue by ripefig** [maoschanz/drawing#123]
Seems like every drawing app should have this, so you can arrange stuff on the canvas. The main way to do that is to implement support for layers, although you could also do what Google drawing does and basically make every shape it's own layers.

**Comment by maoschanz** [comment]
i mark it as duplicate because it's something mentionned in #32 that i listed in #21, but it'll stay opened since having a dedicated issue might be useful

>The main way to do that is to implement support for layers

no 😁 all the shapes and lines are already in the history, and editing some numbers already stored (probably what Google Drawing does even if i've never tested it) requires far less memory than compositing dozens of actual layers. Layers are cool, powerful and technically "easy" to do but i'm not here to challenge Photoshop or G(l)imp(se), this is more a clone of Microsoft Paint or KolourPaint. I wouldn't even know how to put the layers management controls in the UI

>every drawing app should have this

it's a feature i want to eventually provide, but KolourPaint, XPaint, gPaint, and Wine's MS Paint don't provide it at all so i don't feel like it's really lacking. I mean, it could be here, but its absence doesn't make the app unusable, it'll happen but it's not a priority

**Comment by ripefig** [comment]
Those apps are not unusable but they have a feature set from 1980s. If you look at new applications they all allow you to reposition objects on canvas, in increasingly efficient ways. Nobody would use Google Draw if it didn't allow you to that, or at least I certainly wouldn't.

**Comment by zzzazzz** [comment]
> > every drawing app should have this
> 
> it's a feature i want to eventually provide, but KolourPaint, XPaint, gPaint, and Wine's MS Paint don't provide it at all so i don't feel like it's really lacking. I mean, it could be here, but its absence doesn't make the app unusable, it'll happen but it's not a priority

Right, personaly i love this simple app to edit screenshots and explain things to others by writing on it with my pen wacom. Layers should be a big feature. But the first needed is : **fast to open** !! Krita is great but too slow.

**Comment by zzzazzz** [comment]
> > Layers should be a big feature.
> 
> But this is not the topic of this issue...?

Of course ! right :+1:  :)

**Comment by mbousq** [comment]
Just passing by. I know the [shutter project](https://github.com/shutter-project/shutter) has this. Might be worth checking how it is done there. I also wonder (I am not knowledgeable here) if the solution could be a two-layers based app (two and only two layers, no layers management whatsoever). Workflow would be:

1. Import picture
2. At import, Drawing automatically creates a second layer
3. All edits are done in that second layer,

Point 3. means that we could use the "Rectangle selection" to move things around.

This is basically what I do in KolourPaint, Pinta etc : I always create a second layer before any edits. Potentially automating that at image import is the easy solution.

Just some thoughts! Many thanks for this great app :)

**Comment by maoschanz** [comment]
why do you all keep talking about layers, i already said it's not the topic 

The checklist with everything to do is already written.

>we could use the "Rectangle selection" to move things around

it's not about moving things, it's about editing them.

the user, with the "shape" tool, has drawn a blue rectangle with a 4px line width, and now they think it would look better if it were an orange ellipsis with a 3px line width. The tool they'll use to edit that shape has to provide all the parameters and options the "shape" tool provided when the user drew their original circle: this tool is quite obviously the "shape" tool itself.

**Comment by mbousq** [comment]
> why do you keep talking about layers? i already said it's not the topic

> it's not about moving things, it's about editing them.

Apologies, I was confused by the first paragraph of the Author. Maybe others as well.
 
> Seems like every drawing app should have this, **so you can arrange stuff on the canvas**. The main way to do that is to implement support for layers, although you could also do what Google drawing does and basically make every shape its own layers.
